keras_cv/layers/preprocessing/random_crop.py

```diff
@@ -0,0 +1,132 @@
+      seed: Integer. Used to create a random seed.
+    """
+
+    def __init__(self, height, width, seed=None, bounding_box_format=None, **kwargs):
+        super().__init__(**kwargs, autocast=False, seed=seed, force_generator=True)
+        self.height = height
+        self.width = width
+        self.seed = seed
+        self.auto_vectorize = False
+        self.bounding_box_format = bounding_box_format
+
+    def call(self, inputs, training=True):
+
+        if training:
+            return super().call(inputs, training)
+        else:
+            inputs = self._ensure_inputs_are_compute_dtype(inputs)
+            inputs, metadata = self._format_inputs(inputs)
+            output = inputs
+            # self._resize() returns valid results for both batched and
+            # unbatched
+            output[IMAGES] = self._resize(inputs[IMAGES])
+            if BOUNDING_BOXES in inputs:
+                output[BOUNDING_BOXES] = self._map_fn(
+                    self._smart_resize_bounding_box, inputs[BOUNDING_BOXES]
+                )
+            return self._format_output(output, metadata)
+
+    def get_random_transformation(self, image=None, **kwargs):
+        image_shape = tf.shape(image)
+        h_diff = image_shape[H_AXIS] - self.height
+        w_diff = image_shape[W_AXIS] - self.width
+        dtype = image_shape.dtype
+        rands = self._random_generator.random_uniform([2], 0, dtype.max, dtype)
+        h_start = rands[0] % (h_diff + 1)
+        w_start = rands[1] % (w_diff + 1)
+        return {"top": h_start, "left": w_start}
+
+    def augment_image(self, image, transformation, **kwargs):
+        image_shape = tf.shape(image)
+        h_diff = image_shape[H_AXIS] - self.height
+        w_diff = image_shape[W_AXIS] - self.width
+        return tf.cond(
+            tf.reduce_all((h_diff >= 0, w_diff >= 0)),
+            lambda: self._crop(image, transformation),
+            lambda: self._resize(image),
+        )
+
+    def _crop(self, image, transformation):
+        top = transformation["top"]
+        left = transformation["left"]
+        return tf.image.crop_to_bounding_box(image, top, left, self.height, self.width)
+
+    def _resize(self, image):
+        resizing_layer = tf.keras.layers.Resizing(self.height, self.width)
+        outputs = resizing_layer(image)
+        # smart_resize will always output float32, so we need to re-cast.
+        return tf.cast(outputs, self.compute_dtype)
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation=None, **kwargs):
+        # Divya to-do add support in follow up PR
+        return bounding_boxes
+
+    def compute_output_shape(self, input_shape):
+        input_shape = tf.TensorShape(input_shape).as_list()
+        input_shape[H_AXIS] = self.height
+        input_shape[W_AXIS] = self.width
+        return tf.TensorShape(input_shape)
+
+    def get_config(self):
```

Comment:
Let's add this layer to layers/serialization_test.py

Reply:
Done!